homeassistant/components/zwave_js/config_flow.py

```diff
@@ -62,14 +110,268 @@ async def async_step_user(
+        assert self.hass
+        if not self.install_task:
+            self.install_task = self.hass.async_create_task(self._async_install_addon())
+            return self.async_show_progress(
+                step_id="install_addon", progress_action="install_addon"
+            )
+
+        assert self.hass
+        try:
+            await self.install_task
+        except self.hass.components.hassio.HassioAPIError as err:
+            _LOGGER.error("Failed to install Z-Wave JS add-on: %s", err)
+            return self.async_show_progress_done(next_step_id="install_failed")
+
+        self.integration_created_addon = True
+
+        return self.async_show_progress_done(next_step_id="start_addon")
+
+    async def async_step_install_failed(
+        self, user_input: Optional[Dict[str, Any]] = None
+    ) -> Dict[str, Any]:
+        """Add-on installation failed."""
+        return self.async_abort(reason="addon_install_failed")
+
+    async def async_step_start_addon(
+        self, user_input: Optional[Dict[str, Any]] = None
+    ) -> Dict[str, Any]:
+        """Ask for config and start Z-Wave JS add-on."""
+        if self.addon_config is None:
+            self.addon_config = await self._async_get_addon_config()
+
+        errors = {}
+
+        if user_input is not None:
+            self.network_key = user_input[CONF_NETWORK_KEY]
+            self.usb_path = user_input[CONF_USB_PATH]
+
+            new_addon_config = {
+                CONF_ADDON_DEVICE: self.usb_path,
+                CONF_ADDON_NETWORK_KEY: self.network_key,
+            }
+
+            if new_addon_config != self.addon_config:
+                await self._async_set_addon_config(new_addon_config)
+
+            assert self.hass
+            try:
+                await self.hass.components.hassio.async_start_addon("core_zwave_js")
+            except self.hass.components.hassio.HassioAPIError as err:
+                _LOGGER.error("Failed to start Z-Wave JS add-on: %s", err)
+                errors["base"] = "addon_start_failed"
+            else:
+                # Sleep some seconds to let the add-on start properly before connecting.
+                await asyncio.sleep(ADDON_SETUP_TIME)
+                discovery_info = await self._async_get_addon_discovery_info()
+                self.ws_address = (
+                    f"ws://{discovery_info['host']}:{discovery_info['port']}"
+                )
+
+                if not self.unique_id:
+                    try:
+                        version_info = await async_get_version_info(
+                            self.hass, self.ws_address
+                        )
+                    except CannotConnect:
+                        return self.async_abort(reason="cannot_connect")
+                    await self.async_set_unique_id(
+                        version_info.home_id, raise_on_progress=False
+                    )
+
+                self._abort_if_unique_id_configured()
```

Comment:
This part is a copy of the corresponding part in `async_step_on_supervisor`. I can make a helper for this to make it more DRY, but wanted to wait for some feedback first in case we change other things in this flow.